src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -615,10 +641,39 @@ private IncrementalHash PrepareDigest()
                         }
                     }
 
-                    writer.PopSetOf();
-                    Helpers.DigestWriter(hasher, writer);
+                    if (compatMode)
+                    {
+                        writer.PopSequence();
+
+#if netcoreapp
+                        Span<byte> setOfTag = stackalloc byte[1];
+                        setOfTag[0] = 0x31;
+
+                        hasher.AppendData(setOfTag);
+                        hasher.AppendData(writer.EncodeAsSpan().Slice(1));
+#else
+                        byte[] encoded = hasher.Encode();
+                        encoded[0] = 0x31;
+                        hasher.AppendData(encoded);
```

Comment:
I'm obviously missing something in the logic here... why is `writer` written out in the netcoreapp version but not here?